### PR TITLE
feat: allow toggling of password fields

### PR DESCRIPTION
**Pull request by hwkr**
Currently I just enabled it for all password fields but we can turn it into a prop pretty easy if that's preferred.

https://github.com/scalar/scalar/assets/6374090/9ed6adc5-ec85-4b23-b7f3-d4d63137ffff

### CODE REVIEW

File: packages/api-client/src/components/ApiClient/Request/RequestAuthentication/CardForm.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
-  <div class="card-form">
+  <form class="card-form">
```
Comment (hwkr) [inline packages/api-client/src/components/ApiClient/Request/RequestAuthentication/CardForm.vue:2]: Chrome was upset this wasn't a `<form>`

Google Chrome-2024-05-02-15-41-01@2x

**Review by hanspagel — APPROVED** [review]
So slick

**Review by amritk — APPROVED** [review]
LGTM! Surprised we never had those icons

File: packages/api-client/src/components/ApiClient/Request/RequestAuthentication/CardFormTextInput.vue [hunk 2]
```diff
@@ -17,8 +30,15 @@ defineOptions({
       v-bind="$attrs"
       :id="id"
       autocomplete="off"
+      data-1p-ignore
```
Comment (antlio) [inline packages/api-client/src/components/ApiClient/Request/RequestAuthentication/CardFormTextInput.vue:33]: 🙏

**Review by marclave — APPROVED** [review]
omg the rick roll  😆 never gunna give you up 😆 

nice work @hwkr

**Pull request merged**